tests/rptest/tests/full_disk_test.py

```diff
@@ -346,22 +347,28 @@ def test_target_min_capacity_wanted_time_based(self):
         reported = admin.get_local_storage_usage(
             node)["target_min_capacity_wanted"]
 
-        # params. the size is about 900k larger than what was written,
-        # attributable to per record overheads etc... and determined emperically
-        # by looking at trace log stats.
+        # The size is slightly larger than what was written, attributable to
+        # per record overheads, indices, fallocation, etc... The expected size
+        # is determined empirically by looking at trace log stats.
         size = 32664482
         time = 61
         retention = 3600
         expected = retention * (size / time)
 
-        # factor in the 2 segments worth of space for controller log
-        diff = abs(reported - expected - 2 * default_segment_size)
+        # Subtract the controller segment size from the expected size as we
+        # are only interested in the topic under test data size.
+        diff = reported - controller_segment_size - expected
```

Comment:
> Slightly improve readability of the test.

it looks like the calculation changed, too? is it related to how the pr fixes the test failure?

Reply:
[Reverted few things back](https://github.com/redpanda-data/redpanda/compare/aec95483ce1b09070f29cb37d5f8d798dcad0cf3..dbeceb1366f7f8498fd6c7d0f657f1331e8c0bb8). This was a non-functional change but still was incorrect. Controller log was reporting 256 MiB wanted with one segment and I thought that matches the compacted segment size so it must be it (2x default segment size). But that logic was incorrect. Moreover, it should have been clamped anyway by max segment size.

Updated the logic to properly match what is happening in disk_log_impl. In particular, for log without cleanup policy we 2x it's current size and multiply `storage_reserve_min_segments` by default segment size and take the maximum.

There still shouldn't be any changes. The code changed but all the numbers stay the same.

Reply:
dump that into the commit message.

Reply:
Updated and rebased on dev.